Paladin/SourceControl/SourceControl.cpp

```diff
@@ -299,39 +308,67 @@ int
 SourceControl::RunCommand(BString in, BString &out)
 {
 	if (fDebug)
-		printf("Command: %s: %s\n", fShortName.String(), in.String());
+		STRACE(2,("Command: %s: %s\n", fShortName.String(), in.String()));
 	
 	if (in.CountChars() < 1)
 		return -1;
 	
-	in << "\necho \"Source control command return value: $?\"";
-	
+	//in << "\necho \"Source control command return value: $?\"";
+	/*
 	FILE *fd = popen(in.String(),"r");
 	
 	if (!fd)
 		return -2;
+	*/
 	
 	out = "";
+	
+	/*
 	char buffer[1024];
 	while (fgets(buffer,1024,fd))
 	{
-		if (fCallback)
-			fCallback(buffer);
-		out += buffer;
+		if (!ferror(fd)) {
+			if (fCallback)
+				fCallback(buffer);
+			out += buffer;
+		} else {
+			STRACE(2,("Command response buffer in error for SCM command"));
+		}
 	}
 	
-	pclose(fd);
+	int result = pclose(fd);
+	*/
+	BString cmd(in);
+	//cmd << "sh -c \"" << in << "\"";
+	STRACE(2,("SourceControl::RunCommand:Command: %s\n",cmd.String()));
+	status_t retval = RunPipedCommand(cmd.String(),out,true);
+	STRACE(2,("Command complete\n"));
+	int result = 0;
+	if (B_OK != retval) 
+		result = -1;
 	
 	if (fDebug)
-		printf("%s: out:\n------------\n%s------------\n",
-				GetShortName(), out.String());
+		STRACE(1,("%s: out:\n------------\n%s------------\n",
+				GetShortName(), out.String()));
 	
 	// Grab the return value of the command
-	int32 pos = out.FindLast("Source control command return value");
+	//int32 pos = out.FindLast("Source control command return value");
+	
+	//int returnValue = atoi(out.String() + pos + strlen("Source control command return value: "));
+	//out.Truncate(pos);
+	out << "----------\n";
+	if (-1 == result) {
+		out << B_TRANSLATE("Command resulted in an error.\n");
+	} else {
+		out << 	B_TRANSLATE("Command succeeded. Use 'Import existing Project' "
```

Comment:
Shouldn't that be 'Import existing project'. Sentence-casing is used everywhere else.

Reply:
Yes very true. I missed that one. Feel free to fix if you have time. Otherwise, please lot as a bug.

Reply:
How about I fix that and newly generated en.catkeys which I upload at Polyglot?
Or do you expect major changes in the GUI strings? Additional or removed strings aren't a problem, only changed strings, as those have to be translated from scratch again.

Reply:
I don't think any will really change. I think there may still be some that don't yet exist. Feel free to upload new en.catkeys at polyglot and translate. Thanks!